com.unity.netcode.gameobjects/Runtime/Serialization/IBufferSerializerImplementation.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    public interface IBufferSerializerImplementation
```

Comment:
can't it just be `IBufferSerializer`?

Reply:
I think that'd be confusing since `BufferSerializer` doesn't implement `IBufferSerializer`, but wraps it.

Reply:
it is still confusing either way :D "Implementation" doesn't communicate anything extra here, it looks arbitrary.
and, BufferSerializer can do whatever it wants, I wouldn't change something because of something else doing something different — if `IBufferSerializer` is more sound, let's use it. if `BufferSerializer<T>` is confusing, it is still confusing, I don't get it :P

Reply:
Renamed it to IReaderWriter. BufferSerializer<IBufferSerializer> just doesn't sit right with me. BufferSerializer<IReaderWriter> is a bit more clear, IMO.